src/App.tsx

```diff
@@ -15,48 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const visibleGoods = [...goodsFromServer];
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortBy, setSortBy] = useState(SortType.NONE);
+
+  function reset() {
+    setIsReversed(false);
+    setSortBy(SortType.NONE);
+  }
+
+  if (sortBy === SortType.ALPHABET) {
+    visibleGoods.sort();
+  }
+
+  if (sortBy === SortType.LENGTH) {
+    visibleGoods.sort((good1, good2) => {
+      return good1.length - good2.length;
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortBy(SortType.ALPHABET)}
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            {
+              'button is-info is-light':
+                sortBy !== SortType.ALPHABET,
+            },
+            {
+              'button is-info':
+                sortBy === SortType.ALPHABET,
+            },
+          )}
```

Comment:
If I understood you correctly, you want to always have some styles and add only one if necessary, then it is better to write it like this

```suggestion
          className={classNames('button', 'is-info', 
            {
              'is-light':
                sortBy !== SortType.ALPHABET,
            },
          )}
```

Reply:
Yes thanks